src/Transactions/Builder/AbstractTransactionBuilder.php

```diff
@@ -35,7 +35,7 @@ public static function new(?array $data = null): static
         return new static($data);
     }
 
-    public function gasLimit(int $gasLimit): static
+    public function gasLimit(int | string $gasLimit): static
```

Comment:
do you think it would be more safe if this method takes a bignumber equivalent as parameter? that means the converter should return a bignumber instance as well. Added benefit is that the caller can then decide how to use the bignumber values instead of the sdk forcing them into a specific format

Reply:
I've updated this. Once you're happy, I'll update the docs PR :oleglala: